### PR TITLE
streamer: add stake quota overflow tests

### PR DESCRIPTION
**Pull request by stablebits**
#### Problem
The stream throttling tests don't exercise mainnet-scale stake values, which is how the u64 overflow in the staked stream quota computation (fixed in #14799) went unnoticed.

#### Summary of Changes
- Rework the staked-capacity tests to use mainnet-scale stakes, with cases below and above the u64 multiplication-overflow boundary (with and without unstaked connections allowed).
- Add a test for maximum (`u64::MAX`) stake values.

This is a draft until #14799 lands (otherwise tests fail). To be rebased and marked ready afterwards.

### CODE REVIEW

**Comment by alexpyattaev** [comment]
rebase?

File: streamer/src/nonblocking/stream_throttle.rs [hunk 1]
```diff
@@ -281,6 +281,25 @@ pub mod test {
         std::sync::{Arc, atomic::Ordering},
     };
 
+    const LAMPORTS_PER_SOL: u64 = 1_000_000_000;
```
Comment (efagerho) [inline streamer/src/nonblocking/stream_throttle.rs:284]: Are we not exposing this as a public constant anywhere else?

Comment (alexpyattaev) [reply streamer/src/nonblocking/stream_throttle.rs:284]: AFAIK we are not 🥹

**Review by alexpyattaev — APPROVED** [review]
